presto-common/src/main/java/com/facebook/presto/common/block/AbstractArrayBlock.java

```diff
@@ -74,8 +74,8 @@ public final Block copyPositions(int[] positions, int offset, int length)
 
         int totalElements = newOffsets[length];
         if (totalElements == 0) {
-            // No elements selected
-            Block newValues = getRawElementBlock().copyRegion(getOffsetBase(), 0);
+            // No elements selected, copy a zero-length region from the elements block
+            Block newValues = getRawElementBlock().copyRegion(getOffset(0), 0);
```

Comment:
Curious why do we need to copy a zero-length region. Can't we just make an empty block?

Reply:
You are correct, changed this to `copyRegion(0, 0)` which should always be valid.

Reply:
Or was your suggestion just to remove the special case all together and let it fall through to a zero element copyPositions call below?

Reply:
What you have works. Thanks.